app/controllers/home_controller.rb

```diff
@@ -6,18 +6,20 @@ def home
       @dogs = Dog.for_adoption
       @featured_dog = Dog.featured
 
-      if params['gender']
+      if params['gender'] && params[:gender] != 'any'
```

Comment:
You should be consistent in how you access the params hash.  Either use strings or symbols but don't mix in the same line.  Do you know what I mean here? (and on similar lines below)